File: steps/7/test/scala/eightyDays/scala/BankTest.scala
package eightyDays.scala

import java.util.UUID

import org.scalatest.WordSpec

class BankTest extends WordSpec {
  val have = afterWord("have")

  "A new bank" must have {
    "a given name" in {
      val testee = new Bank("Test bank")
      assert(testee.name === "Test bank")
    }
    "now partners" in {
      val testee = new Bank("Test bank")
      assert(None === testee.person(UUID.randomUUID()))
    }
    "one customer after adding one" in {
      val testee = new Bank("Test bank")
      val fogg = Person("Phileas", "Fogg")

      val phileas = testee.addPartner(fogg)

      assert(Some(fogg) === testee.person(phileas))
      assert(phileas === testee.addPartner(fogg))
    }
  }
}

File: steps/3/test/java/eightyDays/java/BankTest.java
package eightyDays.java;

import org.junit.Test;

import java.util.Optional;
import java.util.UUID;

import static org.junit.Assert.*;

public class BankTest {
    @Test
    public void getName() throws Exception {
        Bank testee = new Bank("TestBank");
        assertEquals("TestBank", testee.getName());
    }

    @Test
    public void getPartner() throws Exception {
        Bank testee = new Bank("TestBank");
        assertNull(testee.getPerson(UUID.randomUUID()));
    }

    @Test
    public void testAddCustomer() {
        Bank testee = new Bank("TestBank");
        Person fogg = new Person("Phileas", "Fogg");

        UUID phileas = testee.addPartner(fogg);
        assertEquals(fogg, testee.getPerson(phileas));
        assertEquals(phileas, testee.addPartner(fogg));
    }
}